apps/arweave/src/ar_info.erl

```diff
@@ -0,0 +1,100 @@
+%%%
+%%% @doc Gathers the data for the /info and /recent endpoints.
+%%%
+
+-module(ar_info).
+
+-export([get_info/0, get_recent/0]).
+
+-include_lib("arweave/include/ar.hrl").
+-include_lib("arweave/include/ar_chain_stats.hrl").
+
+get_info() ->
+	{Time, Current} =
+		timer:tc(fun() -> ar_node:get_current_block_hash() end),
+	{Time2, Height} =
+		timer:tc(fun() -> ar_node:get_height() end),
+	[{_, BlockCount}] = ets:lookup(ar_header_sync, synced_blocks),
+    #{
+        <<"network">> => list_to_binary(?NETWORK_NAME),
+        <<"version">> => ?CLIENT_VERSION,
+        <<"release">> => ?RELEASE_NUMBER,
+        <<"height">> =>
+            case Height of
+                not_joined -> -1;
+                H -> H
+            end,
+        <<"current">> =>
+            case is_atom(Current) of
+                true -> atom_to_binary(Current, utf8);
+                false -> ar_util:encode(Current)
+            end,
+        <<"blocks">> => BlockCount,
+        <<"peers">> => prometheus_gauge:value(arweave_peer_count),
+        <<"queue_length">> =>
+            element(
+                2,
+                erlang:process_info(whereis(ar_node_worker), message_queue_len)
+            ),
+        <<"node_state_latency">> => (Time + Time2) div 2
+    }.
+
+get_recent() ->
+    #{
+        %% #{
+        %%   "id": <indep_hash>,
+        %%   "received": <received_timestamp>"
+        %% }
+        <<"blocks">> => get_recent_blocks(),
+        %% #{
+        %%   "id": <hash_of_block_ids>,
+        %%   "height": <height_of_first_orphaned_block>,
+        %%   "timestamp": <timestamp_of_when_fork_was_abandoned>
+        %%   "blocks": [<block_id>, <block_id>, ...]
+        %% }
+        <<"forks">> => get_recent_forks()
+    }.
+
+get_recent_blocks() ->
+    lists:foldl(
+        fun(H, Acc) ->
+            Acc ++ [#{
+                <<"id">> => ar_util:encode(H),
+                <<"received">> => get_block_timestamp(H, length(Acc))
+            }]
+        end,
+        [],
+        lists:sublist(ar_node:get_block_anchors(), ?CHECKPOINT_DEPTH)
+    ).
+
+get_recent_forks() ->
+    CutOffTime = os:system_time(seconds) - ?RECENT_FORKS_AGE,
+    case ar_chain_stats:get_forks(CutOffTime) of
+        {error, _} -> error;
+        Forks ->
+            lists:foldl(
+                fun(Fork, Acc) ->
+                    #fork{ 
+                        id = ID, height = Height, timestamp = Timestamp, 
+                        block_ids = BlockIDs} = Fork,
+                    Acc ++ [#{
```

Comment:
20 times is not a lot, but I would avoid some unnecessary reductions here (e.g., might matter slightly under load) and prepend `[#{...} | Acc]`, with lists:reverse if we want this ordering.

Reply:
Updated! https://github.com/ArweaveTeam/arweave/pull/581/commits/9e18942770c521b4dcb50e31cc00641a2cdea790